server/testnet.go

```diff
@@ -0,0 +1,240 @@
+/*
+ * Ported from Tendermint
+ */
+package server
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	cfg "github.com/tendermint/tendermint/config"
+	gaiacfg "github.com/cosmos/cosmos-sdk/config"
+	"github.com/tendermint/tendermint/p2p"
+	"github.com/tendermint/tendermint/types"
+	pvm "github.com/tendermint/tendermint/types/priv_validator"
+	cmn "github.com/tendermint/tmlibs/common"
+	"github.com/cosmos/cosmos-sdk/wire"
+)
+
+var (
+	nValidators    int
+	nNonValidators int
+	outputDir      string
+	nodeDirPrefix  string
+
+	populatePersistentPeers bool
+	hostnamePrefix          string
+	startingIPAddress       string
+	p2pPort                 int
```

Comment:
let's not use any global variables whatsoever here - let's pass variables around as necessary